### ISSUE TITLE
AugAssign not support

### ISSUE THREAD

**Issue by jayvdb** [Coronon/PySchemeTranspiler#9]
The following is not supported
```py
sum: int = 0
for i in range(10):
    sum += i
```

**Comment by Coronon** [comment]
```py
sum: int = 0
for i in range(10):
    sum = sum + i
```
would work here, but you probably already know that. This codebase is quite huge (and complex!) for a single person, while it is on my list, it does not have high priority as there is a very simple workaround currently.

**Comment by jayvdb** [comment]
I can implement this simplification in the wrapper I am building.  Regarding complexity, and a single person, if we get enough of the py2many test cases passing that it is semi-usable, I'll try to get my wrapper into `py2many`, which then gets Racket support, and py2many can direct interested users to your project.  And/or once you have a core set of functionality implemented, you might want to contribute a Racket implementation to py2many, where presumably it would benefit from economy of scale.

**Comment by Coronon** [comment]
Little update: Multiple assignments on one line work now, the only problem are those that try to swap elements in lists... For example:
```py
a: List[int] = [1, 2, 3]
b: List[int] = [4, 5, 6]

a[0], b[0] = b[0], a[0]
print("A:", a)
print("B:", b)

>>> A: [4, 2, 3]
>>> B: [1, 5, 6]
```
<details>
  <summary>Racket code generated:</summary>

```racket
#lang racket
(require data/gvector)
(define (safe-gvector-set! vec i elm) (if (< i (gvector-count vec)) (gvector-set! vec i elm) (raise "IndexError: list assignment index out of range" #t)))
(define (gvector-pop! vec i) (define ret (gvector-ref vec i)) (gvector-remove! vec i)ret)
(define (gvector-access vec i) (if (>= i 0) (gvector-ref vec i) (gvector-ref vec (+ (gvector-count vec) i))))
(define (PRINT . args) (for-each (lambda (x i) (unless (= i 0) (display " ")) (display x)) args (range (length args)))(newline))

(define a (gvector 1 2 3))
(define b (gvector 4 5 6))
((lambda () (define __a__ a)(define __b__ b)(safe-gvector-set! a (if (< 0 0) (- (gvector-count a) (- 0)) 0) (gvector-access __b__ 0))(safe-gvector-set! b (if (< 0 0) (- (gvector-count b) (- 0)) 0) (gvector-access __a__ 0))))
(PRINT "A:" a)
(PRINT "B:" b)
```
</details>


Which in racket prints:

```
A: #<gvector: 4 2 3>
B: #<gvector: 4 5 6>
```

As you can see we did not really swap elements in racket. What I do atm is that I create temporary **reference** copys to allow syntax like this (which works with my implementation in racket as well):
```py
a: List[int] = [1, 2, 3]
b: List[int] = [4, 5, 6]

a, b = b, a
print("A:", a)
print("B:", b)

>>> A: [4, 5, 6]
>>> B: [1, 2, 3]
```
<details>
  <summary>Racket code generated:</summary>

```racket
#lang racket
(require data/gvector)
(define (safe-gvector-set! vec i elm) (if (< i (gvector-count vec)) (gvector-set! vec i elm) (raise "IndexError: list assignment index out of range" #t)))
(define (gvector-pop! vec i) (define ret (gvector-ref vec i)) (gvector-remove! vec i)ret)
(define (gvector-access vec i) (if (>= i 0) (gvector-ref vec i) (gvector-ref vec (+ (gvector-count vec) i))))
(define (PRINT . args) (for-each (lambda (x i) (unless (= i 0) (display " ")) (display x)) args (range (length args)))(newline))

(define a (gvector 1 2 3))
(define b (gvector 4 5 6))
((lambda () (define __a__ a)(define __b__ b)(set! a __b__)(set! b __a__)))
(PRINT "A:" a)
(PRINT "B:" b)
```
</details>

The problem is that we would need to create a deep-copy of all lists to satisfy the first use-case... I will have a look into writing a racket extraCode that handles these situations a bit more gracefully (no guarantees though)

PS: We already have to copy all lists and tuples (convert internal type to actual scheme list) when we try to iterate over them and everyone should know scheme is waaayyy slower than Python anyways so this approach does not seem to bad. (I mean, the whole language is designed around returning copys of variables <- Scheme)

For reference: 66b31206bae71af271b7aea40d7f3bdd8afb9aec
Bug fixes: c8381d2eec45abc77f3056fa92f89b87319439d2

PS: I just noticed this is not AugAssign lmao, but your opinion would still be nice :)

**Comment by Coronon** [comment]
AugAssign added in: 42d1337516a482fdab72809676c157b7da1681d9
